### PR TITLE
pos_debt_notebook - unit-tour-test

### PR DESCRIPTION
**Pull request by fedoranvar**

### CODE REVIEW

File: pos_debt_notebook/static/src/js/test_pos_debt.js [hunk 1]
```diff
@@ -59,27 +66,158 @@ odoo.define('pos_debt_notebook.tour', function (require) {
...
+    // Initial Test
+    function initial_scene(steps) {
+        steps = steps.concat(add_product_to_order('Miscellaneous'));
+        steps = steps.concat(click_to_payment());
+        steps = steps.concat(set_customer('Agrolait'));
+        steps = steps.concat(debt_method_paying('Credits (USD)'));
+        steps = steps.concat(validate_order());
+        steps = steps.concat(next_order());
+        return steps;
+    }
```
Comment (KolushovAlexandr) [inline pos_debt_notebook/static/src/js/test_pos_debt.js:140]: for better readability, I can suggest the next:
```
return steps.concat(
    func1(),
    func2(),
    ...
)
```

File: pos_debt_notebook/static/src/js/test_pos_debt.js [hunk 2]
```diff
@@ -59,27 +66,158 @@ odoo.define('pos_debt_notebook.tour', function (require) {
...
+    // Error #1 from https://github.com/it-projects-llc/pos-addons/issues/592
```
Comment (KolushovAlexandr) [inline pos_debt_notebook/static/src/js/test_pos_debt.js:141]: :+1:

Comment (KolushovAlexandr) [reply pos_debt_notebook/static/src/js/test_pos_debt.js:141]: Add here and everywhere short descriptions like **Error: "You cannot use Debt payment. Select customer first."**
so you dont need to open the link

File: pos_debt_notebook/static/src/js/test_pos_debt.js [hunk 3]
```diff
@@ -59,27 +66,158 @@ odoo.define('pos_debt_notebook.tour', function (require) {
...
+    function scene1(steps) {
+        steps = steps.concat(add_product_to_order("Miscellaneous"));
+        steps = steps.concat(click_to_payment());
+        steps = steps.concat(set_customer('Agrolait'));
+        steps = steps.concat(debt_method_paying('Credits (USD)'));
+        steps = steps.concat(deselect_customer());
+        steps = steps.concat(validate_order());
+        steps = steps.concat(close_error_modal_dialog());
```
Comment (KolushovAlexandr) [inline pos_debt_notebook/static/src/js/test_pos_debt.js:149]: make a check for the alert message inside the function

File: pos_debt_notebook/static/src/js/test_pos_debt.js [hunk 4]
```diff
@@ -59,27 +66,158 @@ odoo.define('pos_debt_notebook.tour', function (require) {
...
+        steps = steps.concat(debt_method_paying('Credits (USD)'));
+        steps = steps.concat(validate_order());
+        steps = steps.concat(next_order());
+        return steps;
+    }
+    // Error #1 from https://github.com/it-projects-llc/pos-addons/issues/592
+    function scene1(steps) {
+        steps = steps.concat(add_product_to_order("Miscellaneous"));
+        steps = steps.concat(click_to_payment());
+        steps = steps.concat(set_customer('Agrolait'));
```
Comment (KolushovAlexandr) [inline pos_debt_notebook/static/src/js/test_pos_debt.js:145]: you dont need to select a customer if you deselect it later

File: pos_debt_notebook/static/src/js/test_pos_debt.js [hunk 5]
```diff
@@ -59,27 +66,158 @@ odoo.define('pos_debt_notebook.tour', function (require) {
...
-            content: 'Check proceeded validation',
         }];
     }
+    function close_error_modal_dialog() {
+        return [{
+            trigger: '.modal-dialog .button:contains("Ok")',
+            content: _t("Close alert-dialog"),
+        }];
+    }
+    function return_from_paying_to_menu() {
```
Comment (KolushovAlexandr) [inline pos_debt_notebook/static/src/js/test_pos_debt.js:91]: return_from_paymentscreen_to_menu

File: pos_debt_notebook/static/src/js/test_pos_debt.js [hunk 6]
```diff
@@ -59,27 +66,158 @@ odoo.define('pos_debt_notebook.tour', function (require) {
...
+        steps = steps.concat(return_from_paying_to_menu());
+        steps = steps.concat(click_numpad_mode("Qty"));
+        steps = steps.concat(null_selected_product());
+        return steps;
+    }
+    // Error #4 from https://github.com/it-projects-llc/pos-addons/issues/592
+    function scene4_1(steps) {
+        steps = steps.concat(add_product_to_order("Miscellaneous"));
+        steps = steps.concat(click_numpad_mode("Price"));
+        steps = set_numpad_value(steps,"1000");
```
Comment (KolushovAlexandr) [inline pos_debt_notebook/static/src/js/test_pos_debt.js:159]: you need a value bigger than 1000

File: pos_debt_notebook/static/src/js/test_pos_debt.js [hunk 7]
```diff
@@ -59,27 +66,158 @@ odoo.define('pos_debt_notebook.tour', function (require) {
...
+    function scene4_2(steps) {
+        steps = steps.concat(add_product_to_order("Pay Debt"));
+        steps = steps.concat(click_to_payment());
+        steps = steps.concat(set_customer('Agrolait'));
+        steps = steps.concat(debt_method_paying('Cash (USD)'));
+        steps = set_numpad_value(steps,"1000");
+        steps = steps.concat(debt_method_paying('Credits (USD)'));
+        steps = steps.concat(validate_order());
+        steps = steps.concat(next_order());
+        return steps;
```
Comment (KolushovAlexandr) [inline pos_debt_notebook/static/src/js/test_pos_debt.js:182]: make a check the balance has changed

File: pos_debt_notebook/static/src/js/test_pos_debt.js [hunk 8]
```diff
@@ -59,27 +66,158 @@ odoo.define('pos_debt_notebook.tour', function (require) {
...
+    function scene4_3(steps) {
+        steps = steps.concat(add_product_to_order("Pay Debt"));
+        steps = steps.concat(click_to_payment());
+        steps = steps.concat(set_customer('Agrolait'));
+        steps = steps.concat(debt_method_paying('Credits (USD)'));
+        steps = set_numpad_value(steps,"1000");
+        steps = steps.concat(debt_method_paying('Credits (Fruits & Vegetables only) (USD)'));
+        steps = steps.concat(validate_order());
+        steps = steps.concat(next_order());
+        return steps;
```
Comment (KolushovAlexandr) [inline pos_debt_notebook/static/src/js/test_pos_debt.js:194]: make a check the balance has changed

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]
add error message checks for all functions **close_error_modal_dialog**
two scenarios are missed in the issue

1
* add not fruits and vegetables product to order
* try to pay with fruits and vegetables paymentmethod
* get error

2
process an order using only autopay button (the green one in the bottom left corner)

Don't forget to increase the timeout in python

Odoo version in link http://runbot.it-projects.info/demo/pos-addons/9.0 is not valid
pos_debt_notebook/__init__.py:1: [E7989(readme-versions), ] Wrong version in module's README.rst

File: pos_debt_notebook/static/src/js/test_pos_debt.js (hunk 4 above)
Comment (fedoranvar) [reply pos_debt_notebook/static/src/js/test_pos_debt.js:145]: It requires in test-scenario

We need to check that to pay by Credit, we need to define the customer
But we can't choose credit payment method without at first selecting the customer

File: pos_debt_notebook/static/src/js/test_pos_debt.js (hunk 6 above)
Comment (fedoranvar) [reply pos_debt_notebook/static/src/js/test_pos_debt.js:159]: 1000 was enough in tests
Well, what the heck, so 10 000 will be it

File: pos_debt_notebook/static/src/js/test_pos_debt.js (hunk 4 above)
Comment (KolushovAlexandr) [reply pos_debt_notebook/static/src/js/test_pos_debt.js:145]: Ok :+1:

**Pull request closed without merging**